cli/src/main/java/com/devonfw/tools/ide/environment/AbstractEnvironmentVariables.java

```diff
@@ -181,35 +214,56 @@ private String resolve(String value, Object src, int recursion, Object rootSrc,
     StringBuilder sb = new StringBuilder(value.length() + EXTRA_CAPACITY);
     do {
       String variableName = matcher.group(2);
-      String variableValue = getValue(variableName);
-      if (variableValue == null) {
-        this.context.warning("Undefined variable {} in '{}={}' for root '{}={}'", variableName, src, value, rootSrc,
-            rootValue);
+      String variableValue = "";
+      if (src.toString().equals(variableName)) {
```

Comment:
We can not rely on `src` this way if you read the contract (JavaDoc) of the `resolve` method because `src` can be anything and is only for debugging and error reporting.

Reply:
Thansk I didn't know this. I fixed it.